packages/web-runtime/src/services/auth/userManager.ts

```diff
@@ -284,6 +286,52 @@ export class UserManager extends OidcUserManager {
     this.store.commit('SET_CAPABILITIES', capabilities)
   }
 
+  // copied from upstream oidc-client-ts UserManager with CERN customization
+  protected async _signinEnd(url: string, verifySub?: string): Promise<User> {
+    if (!this.configurationManager.options.isRunningOnEos) {
+      return super._signinEnd(url, verifySub)
+    }
+
+    const logger = this._logger.create('_signinEnd')
+    const signinResponse = await this._client.processSigninResponse(url)
+    logger.debug('got signin response')
+
+    const user = new User(signinResponse)
+    if (verifySub) {
+      if (verifySub !== user.profile.sub) {
+        logger.debug(
+          'current user does not match user returned from signin. sub from signin:',
+          user.profile.sub
+        )
+        throw new ErrorResponse({ ...signinResponse, error: 'login_required' })
+      }
+      logger.debug('current user matches user returned from signin')
+    }
+
+    /* CERNBox customization
```

Comment:
Did you discuss adding a kind of `post login hook` extension entrypoint? It could receive the loaded user as param and then `do stuff` ™️  / modify the user, etc. I can think of other useful scenarios for this... but at the same time my assumption is, that it's too early for such an extension entrypoint.

Reply:
As discussed: let's not do this. It's super-non-standard and I don't want an "official extension point". If you don't need to mess with the token itself or its expiration time, you can just use events that are triggered by the userManager...